index.bs

```diff
@@ -1895,14 +1874,15 @@ const descB = {type: 'float32', dimensions: [4, 3]};
 const bufferB = new Float32Array(sizeOfShape(descB.dimensions)).fill(0.5);
 const b = builder.constant(descB, bufferB);
 const c = builder.matmul(a, b);
-const graph = await builder.build({c});
+const graph = await builder.build({'c': c});
 
 const bufferA = new Float32Array(sizeOfShape(descA.dimensions)).fill(0.5);
 const inputs = {'a': {data: bufferA}};
 // Pre-allocate output buffer for c.
-const outputs = {'c': {data: new Float32Array(sizeOfShape([3, 3]))}};
-await graph.compute(inputs, outputs);
-console.log(&#96;values: ${outputs.c.data}&#96;);
+const bufferC = new Float32Array(sizeOfShape([3, 3]));
+const outputs = graph.compute(inputs);
+await outputs.c.data(bufferC);
```

Comment:
This pattern implies that for every `compute` call, the browser must allocate an intermediate resource for the execution result, since the only pre-allocated output buffer allowed is the buffer used as the download destination. I'm thinking in the case of an implementation on the GPU where the execution result will need to be staged in a GPU buffer that can't be pre-allocated according to this change.

Reply:
Thanks @wchao1115 . I think you are right, the pre-allocated output buffers should be provided at compute call. I revert the `MLOutput` for the pre-allocated output buffers and only for that usage. By following the discussion in #162, I add the `MLTensor` interface for the newly allocated output buffers that are returned by compute call. Please take another look.